Sources/Xcodeproj/Module+PBXProj.swift

```diff
@@ -259,6 +259,23 @@ extension XcodeModuleProtocol  {
             buildSettings["OTHER_SWIFT_FLAGS"] = (["$(inherited)"] + pkgArgs.cFlags).joined(separator: " ")
         }
 
+        // Generate modulemap for a ClangModule if not provided by user and add to build settings.
+        if case let clangModule as ClangModule = self where clangModule.type == .library {
+            buildSettings["DEFINES_MODULE"] = "YES"
+            let moduleMapPath: String
+            // If user provided the modulemap no need to generate.
+            if clangModule.moduleMapPath.isFile {
+                moduleMapPath = clangModule.moduleMapPath
+            } else {
+                // Generate and drop the modulemap inside Xcodeproj folder.
+                let path = Path.join(xcodeProjectPath, "GeneratedModuleMap", clangModule.c99name)
+                try clangModule.generateModuleMap(inDir: path)
+                moduleMapPath = Path(Path.join(path, clangModule.moduleMap)).relative(to: xcodeProjectPath.parentDirectory)
+            }
+
+            buildSettings["MODULEMAP_FILE"] = moduleMapPath
```

Comment:
It does convert to relative path if we generate modulemap (line 273)
Should it also do it if user provides the modulemap?